srml/babe/src/lib.rs

```diff
@@ -345,15 +372,24 @@ impl<T: Trait + staking::Trait> session::OneSessionHandler<T::AccountId> for Mod
 	}
 }
 
+// compute randomness for a new epoch. rho is the concatenation of all
+// VRF outputs in the prior epoch.
+//
+// an optional size hint as to how many VRF outputs there were may be provided.
 fn compute_randomness(
 	last_epoch_randomness: [u8; RANDOMNESS_LENGTH],
 	epoch_index: u64,
-	rho: [u8; VRF_OUTPUT_LENGTH],
+	rho: impl Iterator<Item=[u8; VRF_OUTPUT_LENGTH]>,
+	rho_size_hint: Option<usize>,
```

Comment:
I thought `usize` is not allowed due to different size between wasm and native build?

Reply:
It is not allowed to be shared between `WASM` and native, but otherwise you can use it. The size of a vector can not be greater than `usize::max` anyway.

Reply:
yes, that's right -- this isn't exposed anywhere and it's only used to do memory allocation (in which case it is the right unit)

Reply:
Interesting, can wasm's usize differ when running on different platforms?  If so, we might advise only to run on 64 bit platforms, so no bad parachain code can leak `usize::MAX` somehow.

Reply:
Wasm is currently always only 32bit. Rho can only store 2^32 elments at max.